### PR TITLE
Telemetry

### LINKED ISSUE #415: Collect Telemetry data
We want to collect some informations about vscode-xml usage. Let's start with some easy ones :

- lemminx version
- server mode (JVM vs Binary)
- preferBinary (true/false)
- VM name (Oracle/OpenJDK/GraalVM...) + version

Follow the instructions from https://github.com/redhat-developer/vscode-commons/blob/master/INSTRUCTIONS.md

### PR DESCRIPTION
**Pull request by datho7561**
Implement telemetry. Please refer to `USAGE_DATA.md` to see which data is collected.
Closes #415

### CODE REVIEW

File: package.json
Comment (fbricon) [inline package.json:37]: why removing it?

Comment (datho7561) [reply package.json:37]: The `package-lock.json` format changed in npm 7, so it npm-force-resolutions doesn't work correctly: https://stackoverflow.com/a/65014694.

I shouldn't have made this change, since this likely will cause issues on Jenkins. I'll downgrade my npm so that it will work on my machine.

**Comment by datho7561** [comment]
Going to rebase onto #421 once its merged, marking this PR as ready for review.

File: src/telemetry.ts
Comment (fbricon) [inline src/telemetry.ts:40]: track

**Comment by rgrunber** [comment]
I've observed that if I replace the [default segment](https://github.com/redhat-developer/vscode-commons/blob/master/package.json#L9) key with the debug one (ie. always use debug), rebuild vscode-commons, and install it along with this vscode-xml change then events are always logged, and it's working as expected.

**Comment by fbricon** [comment]
> I've observed that if I replace the [default segment](https://github.com/redhat-developer/vscode-commons/blob/master/package.json#L9) key with the debug one (ie. always use debug), rebuild vscode-commons, and install it along with this vscode-xml change then events are always logged, and it's working as expected.
> 
> If not, I only seem to see events recorded when I'm debugging.

Yes, as explained in https://github.com/redhat-developer/vscode-commons/blob/master/INSTRUCTIONS.md#optional-add-a-custom-segment-key-in-packagejson-file

**Comment by fbricon** [comment]
Can we track binary downloads (and failures)?

**Comment by rgrunber** [comment]
I think we can merge this when ready. I can confirm that success/failure do get logged.

**Comment by datho7561** [comment]
Okay. I am ready to merge if you approve the PR.

**Review by rgrunber — APPROVED** [review]
Looks good. Let's get this in.

**Pull request merged**
